File: 58-length-of-last-word/length-of-last-word.js
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLastWord = function(s) {
    
    /*var max = 0;
    
    //could loop through the string breaking it each time there is a space character found
    var arr = s.split(/\s+/).filter(word => word !== '');

    var last = arr[arr.length - 1]
    //get last that 
    max = last.length;

    return max;
    */
    let t = s.trim();//get rid of the spaces that aren't in the middel

    //create substrings of the final word by finding the last space
    //get the length of that substring
    return t.substring(t.lastIndexOf(" ")+1).length;



};

File: 58-length-of-last-word/length-of-last-word.java
class Solution {
    public int lengthOfLastWord(String s) {
        
        //trim so we can get rid of the extra white space

        //get the last index of " "
        //return the length of the trimed string - the last index -1

        String stimmed = s.trim();

        int lastIndex = stimmed.lastIndexOf(" ");
        
        int len = stimmed.length() - lastIndex -1;
        
        return len;

    }
}